### PR TITLE
swtich to nearest surviving branch after modify

### PR DESCRIPTION
**Pull request by skarim**
After `gh stack modify` applies changes, the user may end up on an
orphaned branch that is no longer part of the stack — for example, if
their checked-out branch was dropped, folded into another branch, or
renamed. Previously, the code blindly restored the original branch
regardless of whether it still existed in the stack.

Add a `resolveCheckoutBranch` helper that inspects the modify plan and
the post-modify stack to determine the best branch to check out:

  1. Still in stack  → keep the original branch (no-op)
  2. Renamed         → check out the new name
  3. Folded down     → check out the fold target (branch below)
  4. Folded up       → check out the fold target (branch above)
  5. Dropped         → check out the nearest surviving neighbor
                       (prefer above, fall back to below)
  6. Fallback        → topmost branch in the stack

Both `ApplyPlan` and `ContinueApply` (the `--continue` path) now use
this helper instead of unconditionally restoring the original branch.
When the resolved branch differs from the original, a message is
printed so the user knows they've been switched.

The resolution uses the pre-modify snapshot (already persisted in the
state file) to determine original adjacency, so it works correctly even
when multiple branches are removed in the same operation.

Includes 12 new tests:
  - 9 unit tests for resolveCheckoutBranch covering all action types,
    edge cases (topmost dropped, multiple drops, empty stack), and
    the fallback path
  - 3 integration tests verifying ApplyPlan checks out the correct
    branch after drop, fold-down, and rename operations

---

<sub>Stack created with <a href="https://github.com/github/gh-stack">GitHub Stacks CLI</a> • <a href="https://gh.io/stacks-feedback">Give Feedback 💬</a></sub>

### CODE REVIEW

File: internal/modify/apply_test.go [hunk 1]
```diff
@@ -1344,3 +1344,299 @@ func TestApplyPlan_ClearsStateForLocalStack(t *testing.T) {
...
+		Trunk:    stack.BranchRef{Branch: "main"},
+		Branches: []stack.BranchRef{{Branch: "A"}, {Branch: "B"}},
+	}
+	snapshot := Snapshot{
+		Branches: []BranchSnapshot{{Name: "A", Position: 0}, {Name: "B", Position: 1}},
+	}
+
+	result := resolveCheckoutBranch("A", nil, snapshot, s)
+	assert.Equal(t, "A", result)
+}
```
[the remark this answers is not in the document]
Comment (skarim) [reply internal/modify/apply_test.go:1361]: It isn't possible to run `modify` from the trunk branch, so this is not a case we need to worry about.

File: internal/modify/apply.go [hunk 2]
```diff
@@ -495,8 +495,13 @@ func ApplyPlan(
...
 
-	// Restore original branch
-	_ = git.CheckoutBranch(currentBranch)
+	// Check out the best branch — the original if it's still in the stack,
+	// otherwise the nearest surviving branch.
+	targetBranch := resolveCheckoutBranch(currentBranch, plan, snapshot, s)
+	_ = git.CheckoutBranch(targetBranch)
+	if targetBranch != currentBranch {
+		cfg.Printf("Switched to %s (original branch %s is no longer in the stack)", targetBranch, currentBranch)
+	}
```
[the remark this answers is not in the document]
Comment (skarim) [reply internal/modify/apply.go:505]: Handled in eb39abc03926399993ede9bda02e13af72025e44

File: internal/modify/apply.go [hunk 3]
```diff
@@ -657,9 +759,13 @@ func ContinueApply(
...
 
-	// All rebases done — restore original branch
+	// All rebases done — check out the best branch
 	if state.OriginalBranch != "" {
-		_ = git.CheckoutBranch(state.OriginalBranch)
+		targetBranch := resolveCheckoutBranch(state.OriginalBranch, state.Plan, state.Snapshot, s)
+		_ = git.CheckoutBranch(targetBranch)
+		if targetBranch != state.OriginalBranch {
+			cfg.Printf("Switched to %s (original branch %s is no longer in the stack)", targetBranch, state.OriginalBranch)
+		}
```
[the remark this answers is not in the document]
Comment (skarim) [reply internal/modify/apply.go:793]: Handled in eb39abc03926399993ede9bda02e13af72025e44

**Review by ktravers — APPROVED** [review]
Very nice UX improvement 🚀
